### DISCUSSION TITLE
What is the way to simulate a rigid diaphragm effect in a 2D analysis?

### DISCUSSION THREAD

**Discussion by PaoloLupo** [TLCFEM/suanPan#239] [Q&A] [answered]
Hi, i'm currently working on a 2D simulation of shearwalls using SGCMQI elements and i need to simulate the effect of a rigid diaphragm, is using the TranslationConnector the correct approach for this?
Thanks

**Comment by TLCFEM** [comment]
Mind to show a brief sketch of the structure?

**Reply by PaoloLupo** [reply]
Sure, here is a sketch of the static analysis I want to perform: 
![Scheme analysis](https://github.com/user-attachments/assets/c9204664-8de0-4f66-ae72-5e12e7503dfb)

**Reply by TLCFEM** [reply, marked as answer]
Okay for such a cantilever wall, typically I won't be too concerned about the rigid floor, because the deformation will be small anyway.
If it is a frame structure then the difference is much larger.

Back to the modelling, you can define relatively rigid beams at each floor, this is effectively penalty method.
Alternatively, it is possible to use [FixedLength](https://tlcfem.github.io/suanPan-manual/latest/Library/Constraint/FixedLength/) constraint to enforce the distance between two nodes stays unchanged.

The [TranslationConnector](https://tlcfem.github.io/suanPan-manual/latest/Library/Element/Special/TranslationConnector/) makes more sense when the nodes do not coincide.
But in this cantilever wall case, [FixedLength](https://tlcfem.github.io/suanPan-manual/latest/Library/Constraint/FixedLength/) should be sufficient.

**Reply by TLCFEM** [reply]
Of course, if you need the _exact_ rigid effect, [TranslationConnector](https://tlcfem.github.io/suanPan-manual/latest/Library/Element/Special/TranslationConnector/) can be used.

For a line with $n$ nodes, define $n-2$ constraints, each uses two end nodes and one of interior nodes.
By such it is effectively enforcing the deformed and un-deformed geometry being identical.
But maybe there is no need for such a strong effect, just a few [FixedLength](https://tlcfem.github.io/suanPan-manual/latest/Library/Constraint/FixedLength/) constraints may suffice.

**Reply by PaoloLupo** [reply]
Thanks, i'm currently using the library for a research project, and it's been really helpful

**Reply by TLCFEM** [reply]
Thanks! Glad to hear it’s helpful. I’m open to any feedback you have.
